pkg/operation/shoot/shoot.go

```diff
@@ -283,6 +283,13 @@ func (b *Builder) Build(ctx context.Context, c client.Client) (*Shoot, error) {
 
 	shoot.Purpose = gardencorev1beta1helper.GetPurpose(shootObject)
 
+	// Determine backup entry name
+	backupEntryName, err := gutil.GenerateBackupEntryName(shootObject.Status.TechnicalID, shootObject.UID)
```

Comment:
Yes, we are sure. The technical ID is computed at https://github.com/gardener/gardener/blob/35db6b62be458890e1c3fabb644a6d52acc435ab/pkg/gardenlet/controller/shoot/shoot_control.go#L477-L480, and this function is called at https://github.com/gardener/gardener/blob/35db6b62be458890e1c3fabb644a6d52acc435ab/pkg/gardenlet/controller/shoot/shoot_control.go#L482.

If it's empty, which can't happen normally but only during "mysterious" circumstances, this will generate an error and the entire flow, including the computation of the technical ID, will be retried.